### ISSUE TITLE
scenarios should support image overrides

### ISSUE THREAD

**Issue by mikedanese** [kubernetes/test-infra#2514]
a la https://github.com/kubernetes/test-infra/blob/master/jenkins/dockerized-e2e-runner.sh#L37

**Comment by fejta** [comment]
That file is dead. What do you want to do? We have canary jobs that allow us to test new images before releasing them.

**Comment by spxtr** [comment]
Our scenarios have lines like this
```python
VERSION_TAG = {
    '1.4': '1.4-v20161130-8958f82',
    '1.5': '1.5-v20161205-d664d14',
    '1.6': '1.6-v20161205-ad918bc',
}
```
that we combine into a `docker run` command like so
```python
check(
    'docker', 'run', '--rm=true', '--privileged=true',
    ...
    'gcr.io/k8s-testimages/kubekins-test:%s' % tag,
)
```

The problem here is that we need to change that tag atomically in a kubernetes/kubernetes PR in order to bump to go 1.8. The change is not backwards-compatible, so we can't switch to the new image before the k8s PR has gone in.

**Comment by krzyzacy** [comment]
you can add `--tag` flag to individual jobs.. And yeah we'll need a go1.8 image for versions that needs go1.8, shall I create one now?

**Comment by spxtr** [comment]
I don't think adding `--tag` helps, since it needs to be done atomically with a k/k change.

**Comment by krzyzacy** [comment]
https://github.com/kubernetes/test-infra/tree/master/jenkins/test-image

We'll need a 1.7 one, does k/k want go 1.8 now? If so I'll make one.

**Comment by spxtr** [comment]
Yes, we'll need a k8s 1.7 job, and it will need to be go 1.8 eventually.
